web/src/compose_validate.ts

```diff
@@ -617,32 +658,41 @@ export function validate_stream_message_address_info(sub: StreamSubscription): b
     return false;
 }
 
-function validate_stream_message(scheduling_message: boolean): boolean {
+function validate_stream_message(scheduling_message: boolean, show_banner: boolean): boolean {
     const stream_id = compose_state.stream_id();
     const $banner_container = $("#compose_banners");
     if (stream_id === undefined) {
```

Comment:
I think it would be a bit more readable to do this instead of calling `set_no_stream_mentioned` in multiple places.

```js
const no_stream_mentioned = stream_id === undefined;
set_no_stream_mentioned(no_stream_mentioned)
if (no_stream_mentioned) {
...
}
```

Same below where you do similar things.